docs: add "Changing the source directory" example (#1286)
denoland/fresh: docs/examples/changing-the-src-dir.md

---
description: |
  Change the source directory to effectively manage your project.
---

When you initialize a project with `deno run -A -r https://fresh.deno.dev`,
you'll end up with a project like the following:

```
.
├── README.md
├── components
│   └── Button.tsx
├── deno.json
├── dev.ts
├── fresh.gen.ts
├── islands
│   └── Counter.tsx
├── main.ts
├── routes
│   ├── [name].tsx
│   ├── api
│   │   └── joke.ts
│   └── index.tsx
└── static
    ├── favicon.ico
    └── logo.svg
```

## Using a `src` directory

> If you'd like your code to live in a `src` directory (or any other place of your
> choosing), then you'll need to do the following things:

Reviewer: ```suggestion
If you'd like your code to live in an `src` directory (or any other directory of your
choosing), then you'll need to do the following things:
```

Author: Ah, the great debate around indefinite articles for abbreviations and acronyms. When I read text like "to live in a `src` directory", I read it as "... in a source directory". I guess you read it as "... in an S. R. C. directory" (i.e. each letter pronounced). I'll do what you say though, since I don't care too much and you're in charge.

> 1. Move most of the code to your chosen directory. For this example I'll stick
>    with `src`.

Reviewer: ```suggestion
1. Move all your files, except `deno.json` and `README.md`, to the `src` directory.
```
For greater clarity. Also, best not to have written in a first-person point of view in the documentation.

Author: yes, sounds better

> 2. Modify your `deno.json` file to point to the new directory.

Reviewer: ```suggestion
2. Modify the `start` task in `deno.json` to point to the new directory.
```
Nits

I initialized a new project, ran `git init` and all the steps to get an
`initial commit` for my fresh Fresh project. Then I made the above changes.
Here's what my `git status` shows:

```
modified:   deno.json
renamed:    components/Button.tsx -> src/components/Button.tsx
renamed:    dev.ts -> src/dev.ts
renamed:    fresh.gen.ts -> src/fresh.gen.ts
renamed:    islands/Counter.tsx -> src/islands/Counter.tsx
renamed:    main.ts -> src/main.ts
renamed:    routes/[name].tsx -> src/routes/[name].tsx
renamed:    routes/api/joke.ts -> src/routes/api/joke.ts
renamed:    routes/index.tsx -> src/routes/index.tsx
renamed:    static/favicon.ico -> src/static/favicon.ico
renamed:    static/logo.svg -> src/static/logo.svg

> ```

Reviewer: This part may not be needed. The file tree below provides a great visualisation of the changes made.

Here's what the diff of `deno.json` looks like:

> ```
> --- a/deno.json
> +++ b/deno.json
> @@ -1,7 +1,7 @@
>  {
>    "lock": false,
>    "tasks": {
> -    "start": "deno run -A --watch=static/,routes/ dev.ts"
> +    "start": "deno run -A --watch=src/static/,src/routes/ src/dev.ts"
>    },
>    "imports": {
>      "$fresh/": "file:///Users/reed/code/fresh/",
> ```

Reviewer: ```suggestion
```diff
 {
   "lock": false,
   "tasks": {
-    "start": "deno run -A --watch=static/,routes/ dev.ts"
+    "start": "deno run -A --watch=src/static/,src/routes/ src/dev.ts"
   },
   "imports": {
     "$fresh/": "file:///Users/reed/code/fresh/",
```
```
This should help highlight differences better.

Author: TIL... thanks!

(An astute observer would note that I actually generated this test project from
my local build of Fresh, via `deno run -A -r ~/code/fresh/init.ts`).

The final result of running `tree` looks like this:

```
.
├── README.md
├── deno.json
└── src
    ├── components
    │   └── Button.tsx
    ├── dev.ts
    ├── fresh.gen.ts
    ├── islands
    │   └── Counter.tsx
    ├── main.ts
    ├── routes
    │   ├── [name].tsx
    │   ├── api
    │   │   └── joke.ts
    │   └── index.tsx
    └── static
        ├── favicon.ico
        └── logo.svg

> ```

Reviewer: Nice!

---
description: |
  Change the source directory to effectively manage your project.
---

When you initialize a project with `deno run -A -r https://fresh.deno.dev`,
you'll end up with a project like the following:

```
.
├── README.md
├── components
│   └── Button.tsx
├── deno.json
├── dev.ts
├── fresh.gen.ts
├── islands
│   └── Counter.tsx
├── main.ts
├── routes
│   ├── [name].tsx
│   ├── api
│   │   └── joke.ts
│   └── index.tsx
└── static
    ├── favicon.ico
    └── logo.svg
```

## Using a `src` directory

If you'd like your code to live in an `src` directory (or any other directory of
your choosing), then you'll need to do the following things:

1. Move all your files, except `deno.json` and `README.md`, to the `src`
   directory.
2. Modify the `start` task in `deno.json` to point to the new directory.

Here's what the diff of `deno.json` looks like:

```diff
 {
   "lock": false,
   "tasks": {
-    "start": "deno run -A --watch=static/,routes/ dev.ts"
+    "start": "deno run -A --watch=src/static/,src/routes/ src/dev.ts"
   },
   "imports": {
     "$fresh/": "file:///Users/reed/code/fresh/",
```

> 
> (An astute observer would note that I actually generated this test project from
> my local build of Fresh, via `deno run -A -r ~/code/fresh/init.ts`).
> 
> The final result of running `tree` looks like this:

Reviewer: ```suggestion
The resulting file structure looks like this:
```
Ah, my bad missed this previously.

Author: Hmmm, ok, I guess, but where do these little snippets go? Once upon a time I didn't know how people were producing these directory structures, so I think it's nice to include a reference to `tree`. Additionally, as I was doing this, it wasn't immediately obvious how to generate a project with my local build. I thought it would be good to include that too.
